components/Typography/tests/unit/MaterialScalableFontTests.m

```diff
@@ -0,0 +1,266 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialTypography.h"
+
+#import "MaterialMath.h"
+#import "../../src/private/UIContentSizeCategory+Material.h"
+
+@interface UIFont_MaterialScalableTests : XCTestCase
+
+@end
+
+@implementation UIFont_MaterialScalableTests
+
+- (void)testScaledFontsReturnEquivalentFonts {
+  // Given
+  NSArray<MDCTextStyle> *textStyles = @[
+    MDCTextStyleHeadline1,
+    MDCTextStyleHeadline2,
+    MDCTextStyleHeadline3,
+    MDCTextStyleHeadline4,
+    MDCTextStyleHeadline5,
+    MDCTextStyleHeadline6,
+    MDCTextStyleSubtitle1,
+    MDCTextStyleSubtitle2,
+    MDCTextStyleBody1,
+    MDCTextStyleBody2,
+    MDCTextStyleButton,
+    MDCTextStyleCaption,
+    MDCTextStyleOverline,
+  ];
+
+  for (MDCTextStyle textStyle in textStyles) {
+    // When
+    UIFont *font1 = [UIFont systemFontOfSize:18.0];
+    UIFont *font2 = [UIFont systemFontOfSize:18.0];
+
+    MDCFontScaler *scaler1 = [[MDCFontScaler alloc] initForMaterialTextStyle:textStyle];
+    MDCFontScaler *scaler2 = [[MDCFontScaler alloc] initForMaterialTextStyle:textStyle];
+
+    UIFont *scaledFont1 = [scaler1 scalableFontWithFont:font1];
+    UIFont *scaledFont2 = [scaler2 scalableFontWithFont:font2];
+
+    // Then
+    XCTAssert([font1 mdc_isSimplyEqual:font2]);
+    XCTAssert([scaledFont1 mdc_isSimplyEqual:scaledFont2]);
+  }
+}
+
+- (void)testScalingCurvesIncrease {
+  // Given
+  NSArray<MDCTextStyle> *textStyles = @[
+    MDCTextStyleHeadline1,
+    MDCTextStyleHeadline2,
+    MDCTextStyleHeadline3,
+    MDCTextStyleHeadline4,
+    MDCTextStyleHeadline5,
+    MDCTextStyleHeadline6,
+    MDCTextStyleSubtitle1,
+    MDCTextStyleSubtitle2,
+    MDCTextStyleBody1,
+    MDCTextStyleBody2,
+    MDCTextStyleButton,
+    MDCTextStyleCaption,
+    MDCTextStyleOverline,
+  ];
+
+  // The following array MUST be ordered from smallest to largest
+  NSArray<UIContentSizeCategory> *sizeCategories = @[
+    UIContentSizeCategoryExtraSmall,
+    UIContentSizeCategorySmall,
+    UIContentSizeCategoryMedium,
+    UIContentSizeCategoryLarge,
+    UIContentSizeCategoryExtraLarge,
+    UIContentSizeCategoryExtraExtraLarge,
+    UIContentSizeCategoryExtraExtraExtraLarge,
+    UIContentSizeCategoryAccessibilityMedium,
+    UIContentSizeCategoryAccessibilityLarge,
+    UIContentSizeCategoryAccessibilityExtraLarge,
+    UIContentSizeCategoryAccessibilityExtraExtraLarge,
+    UIContentSizeCategoryAccessibilityExtraExtraExtraLarge,
+  ];
+
+  for (MDCTextStyle textStyle in textStyles) {
+    // When
+    UIFont *font = [UIFont systemFontOfSize:18.0];
+
+    MDCFontScaler *scaler = [[MDCFontScaler alloc] initForMaterialTextStyle:textStyle];
```

Comment:
Please move this to MDCFontScalerTests - it's not testing the UIFont category methods.  If you intend this to be testing the `mdc_scaledFontForSizeCategory:` API, please mark the FontScaler  portions as `Given` instead of `When`